feat: update readme (#31)
Stelliance/stelliance-widget: README.md

# AngularElementApp

## Technical stack

> Angular element app example.

Reviewer: manque la stack technique

Author: Je rajoute quoi, y'a que les libs à indiquer non?
Used Libraries
- Angular 13
- TypeScript
- Sementic Release
- Unit Test with Karma

Build an angular application into a generic web component that can be uses in any html page.

## Purpose of this library

> Add a menu widget to redirect to the centralized connection of other applications, editable in the configuration file: image, text, links

Reviewer: Added in you frontend, this widget will allow you users to nagivate to other stelliance's companies web applications using SSO.

Author: ok

Author: ok

> # AngularElementApp

Reviewer: ```suggestion
Stelliance widgdet
```

![Last build](https://github.com/Stelliance/stelliance-widget/actions/workflows/build.yml/badge.svg)

## Technical stack

Angular element app example.
Build an angular application into a generic web component that can be uses in any html page.

> Used Libraries

Reviewer: supprimer le texte `Used Libraries`

- Angular 13
- TypeScript

> - Sementic Release

Reviewer: ```suggestion
- Semantic Release
```

- Unit Test with Karma

## Purpose of this library

> Added in you frontend, this widget will allow you users to nagivate to other stelliance's companies web applications using SSO.

Reviewer: ```suggestion
Added in your frontend, this widget will allow your users to navigate to other stelliance's companies web applications using SSO with no re-authentication needed
```

## Serve locally the generic web component

If you need to see the render of the web component locally you can run `npm run preview:serve`.

## How to use it in any frontend

Install the library and declare component in template:

> <stc-app-widget></stc-app-widget>

Reviewer: ca manque de détail, paramètre d'entrée ?

Author: ok

Author: Peux tu me donner le lien vers la lib npm?

## Add github badges in the head of the file (github action status)

> Define actions in .github/workflows

Reviewer: Non je parle d'ajouter un badge dans le readme du projet pour indiquer dans quel etat est le dernier build de la lib.
https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/adding-a-workflow-status-badge

Author: ok je mets le badge last build en haut de page

## Add a link to the github pages url

> https://stelliance.github.io/stelliance-widget/

Reviewer: un lien a href au lien d'un texte en dur ?

Author: ok

## Serve locally the generic web component

If you need to see the render of the web component locally you can run `npm run preview:serve`.

## How to use it in any frontend

Run `npm install <LIB-NAME>` to install the library and declare component in template:

```
<stc-app-widget></stc-app-widget>
```

Inputs:

- `appLogoWidth` : define max width in px allowed for each logo image in template. Default value = `'auto'`

- `appLogoHeight` : define max height in px allowed for each logo image in template. Default value = `'60'`

> - `environment`: `'dev'` | `'prod'` (default value) to use widget-apps-conf.json redirect url in different environments.

Reviewer: ```suggestion
- `environment`: `'dev'` | `'prod'` (default value) to target a specific sso url for each web application
```

> ## Demo Github pages url

Reviewer: ```suggestion
## Demo 
```

> Demo: [Stelliance Widget](https://stelliance.github.io/stelliance-widget/)

Reviewer: ```suggestion
[Stelliance Widget](https://stelliance.github.io/stelliance-widget/)
```
